### PR TITLE
feat(hooks): General cleanup on hooks and pointers

### PR DESCRIPTION
**Pull request by Vali0004**

### CODE REVIEW

File: BigBaseV2/src/hooks/protections/received_clone_sync.cpp [hunk 1]
```diff
@@ -53,14 +47,15 @@ namespace big {
 
 
 		if (invalidsync) {
-
-			if (g->notifications.invalid_sync.log) LOG(WARNING) << "Invalid sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
-			if (g->notifications.invalid_sync.notify) g_notification_service->push_warning("Invalid sync " + std::string(src->get_name()), "Type: " + std::to_string(sync_type) + "\nType name: " + tree_name);
+			if (g->notifications.invalid_sync.log)
+				LOG(WARNING) << "Invalid sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
+			if (g->notifications.invalid_sync.notify)
+				g_notification_service->push_warning(fmt::format("Invalid Sync from {}", src->get_name()), "Type: " + std::to_string(sync_type) + "\nType name: " + tree_name);
```
Comment (Yimura) [inline BigBaseV2/src/hooks/protections/received_clone_sync.cpp]: The first argument you use fmt::format and second argument you don't?

File: BigBaseV2/src/util/math.hpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ namespace big::math
 	inline float calculate_distance_from_game_cam (rage::fvector3 player_position)
 	{
 		const Vector3 plyr_coords = { player_position.x, player_position.y, player_position.z };
-		const Vector3 cam_coords = g_pointers->m_get_gamplay_cam_coords();
+		const Vector3 cam_coords = *g_pointers->m_get_gamplay_cam_coords;
```
Comment (Yimura) [inline BigBaseV2/src/util/math.hpp]: Why do this when it's a function which is supposed to be called?

File: BigBaseV2/src/gta/natives.hpp [hunk 3]
```diff
@@ -54,12 +54,45 @@ namespace rage
...
+			std::uint64_t result;
+			auto charTableOfRegs = (char*)&result - nativeAddress;
+			auto addressIndex = nativeAddress ^ *(DWORD*)(nativeAddress + 8);
+			for (auto i = 0; i < 3; i++) {
+				*(DWORD*)&charTableOfRegs[nativeAddress] = static_cast<DWORD>(addressIndex ^ *(DWORD*)(nativeAddress));
+				nativeAddress += 4;
+			}
+			return result;
+		}
+	};
```
Comment (Yimura) [inline BigBaseV2/src/gta/natives.hpp:95]: What is this, why was it added?

**Review by Yimura — CHANGES_REQUESTED** [review]

File: BigBaseV2/src/util/math.hpp (hunk 2 above)
Comment (Vali0004) [reply BigBaseV2/src/util/math.hpp]: Slightly faster, I guess. More or less unneeded to create it as a function;

File: BigBaseV2/src/gta/natives.hpp (hunk 3 above)
Comment (Vali0004) [reply BigBaseV2/src/gta/natives.hpp:95]: scrNativeRegistration is a subclass of scrNativeRegistrationTable, added it just to have it. In the future, if we wanted, we could auto update our crossmap with it.

File: BigBaseV2/src/hooks/protections/receive_net_message.cpp [hunk 4]
```diff
@@ -1,86 +1,60 @@
...
 		if (frame->get_type() == 4)
 		{
 			rage::datBitBuffer buffer((uint8_t*)frame->m_data, frame->m_length);
 			buffer.m_flagBits = 1;
-			rage::netConnection::MessageType msg_type;
-
+			rage::eNetMessage msgType;
 			const auto player = g_player_service->get_by_msg_id(frame->m_msg_id);
-			if (player && get_message_type(msg_type, buffer))
+			if (player && getMessageType(msgType, buffer))
```
Comment (Yimura) [inline BigBaseV2/src/hooks/protections/receive_net_message.cpp]: Why did you swap to camelCasing, the entire base uses snake_casing.

File: BigBaseV2/src/hooks/protections/receive_net_message.cpp [hunk 5]
```diff
@@ -1,86 +1,60 @@
...
+					case rage::eNetMessage::CMsgTextMessage:
+					{
+						char textBuffer[0x100]{};
+						uint64_t peerId{};
+						bool isTeam{};
+						buffer.ReadString(textBuffer, 0x100);
+						buffer.ReadPeerId(&peerId);
+						buffer.ReadBool(&isTeam);
+						g_notification_service->push_warning("Chat", fmt::format("{} [{}]: {}", player->get_name(), isTeam ? HUD::GET_LABEL_TEXT_("MP_CHAT_TEAM") : HUD::GET_LABEL_TEXT_("MP_CHAT_ALL"), textBuffer));
+					} break;
```
Comment (Yimura) [inline BigBaseV2/src/hooks/protections/receive_net_message.cpp]: Remove this code or add options to toggle this, we don't need this to be permanently enabled.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: BigBaseV2/src/hooks/protections/receive_net_message.cpp (hunk 4 above)
Comment (Vali0004) [reply BigBaseV2/src/hooks/protections/receive_net_message.cpp]: forgor

**Review by Yimura — DISMISSED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**